core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -322,27 +322,20 @@ private RexNode simplifyComparison(RexCall e, RexUnknownAs unknownAs) {
     // Simplify "x <op> x"
     final RexNode o0 = operands.get(0);
     final RexNode o1 = operands.get(1);
-    if (o0.equals(o1)) {
-      RexNode newExpr;
+    if (o0.equals(o1)
```

Comment:
It seems that following changes revert the commit https://github.com/apache/calcite/commit/d10aeb7f7e50dc7028ce102a5f590d0c50c49fa8, I guess they are by mistake?